capstone_project/src/main/java/com/google/sps/data/PostFilter.java

```diff
@@ -0,0 +1,77 @@
+// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.sps.data;
+
+import com.google.appengine.api.datastore.DatastoreService;
+import com.google.appengine.api.datastore.DatastoreServiceFactory;
+import com.google.appengine.api.datastore.Entity;
+import com.google.appengine.api.datastore.FetchOptions;
+import com.google.appengine.api.datastore.Query;
+import com.google.appengine.api.datastore.Text;
+import java.util.ArrayList;
+import java.util.List;
+
+/** Filters and returns all posts that contain the search query. */
+public class PostFilter {
+  public static final DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+
+  private PostFilter() {}
```

Comment:
is having an empty constructor necessary?

Reply:
Since all methods in this class are static, a private constructor prevents the instantiation of the class directly. Having it empty and private means it can't be called outside the class and nothing will happen if it is called.